src/kOS/Suffixed/VesselTarget.cs

```diff
@@ -236,14 +269,41 @@ public ScalarValue GetDistance()
 
         public static string[] ShortCuttableShipSuffixes { get; private set; }
 
+        private int linkCount = 0;
+        public int LinkCount
+        {
+            get
+            {
+                return linkCount; }
+            set
+            {
+                linkCount = value;
+                if (linkCount <= 0)
+                    ScopeLost();
```

Comment:
I've been running with this code in my local copy for a couple weeks now, so I'm fairly confident that nothing ugly can happen.  All that `VesselTarget` does when it looses scope is unhook events, and I did take the time to confirm that there is no issue trying to unsubscribe from an event you aren't subscribed to.